Date.ts

```diff
@@ -1,17 +1,27 @@
+import { isly } from "isly"
 import { DateSpan } from "./DateSpan"
 import { Locale } from "./Locale"
 
 export type Date = string
+//type Digit = "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
+// Representing year better creates to many options
+// export type Date = `${number}-${
+// 	| Exclude<
+// 			`${`0${Exclude<Digit, "0">}` | `1${"0" | "1" | "2"}`}-${Exclude<`${"0" | "1" | "2"}${Digit}`, "00">}` | "30",
+// 			"02-30"
+// 	  >
+// 	| `${"01" | "03" | "05" | "07" | "08" | "10" | "12"}-31`}`
 
 export namespace Date {
-	export function is(value: any | Date): value is Date {
-		return (
-			typeof value == "string" &&
-			value.length == 10 &&
-			new globalThis.Date(value).toString() != "Invalid Date" &&
-			create(new globalThis.Date(value)) == value
+	export const type = isly.named(
+		"isoly.Date",
+		isly.string<Date>(
+			/^(((18|19|20|21|22)[0-9]{2}-(0[13578]|1[02])-(0[1-9]|[12][0-9]|3[01]))|(18|19|20)[0-9]{2}-(0[469]|11)-(0[1-9]|[12][0-9]|30)|(18|19|20)[0-9]{2}-(02)-(0[1-9]|1[0-9]|2[0-8])|(((18|19|20)(04|08|[2468][048]|[13579][26]))|2000)-(02)-29)$/
```

Comment:
This fails for `2400-02-29` 
I'm sure this could be fixed but, I prefer relying on globalThis.Date where they've already implemented all the weird time logic.

Reply:
I can also find other examples like `2296-02-29`

Reply:
Personally, I can't approve of using a big regex to match the dates, because your never gonna get all the leap years right. I would be more ok, with logic that actually checks the year, month, date etc, according to all the rules.

But think it's pretty nice that the Date and Time stuff is just a wrapper around the js-Date-api, that is more usable.
Then we get the added bonus of someone else having done the annoying date and time stuff.

Reply:
Made formula work correctly from 1800 to 2499. There is no way to do this right for everything as 1712-02-30 which is a correct date but only in Sweden. An alternate approach would be to make `isly.string` take a function or use `isly.fromIs` instead.

@EliasEriksson @Magnus93 any suggestions?

Reply:
Added comment on suggestion how the isoly.Date.is should work.
I suggest just using the 4, 100, 400 year rules to check if it's a leap year.

Reply:
```ts
export const type = isly.fromIs<Date>("isoly.Date", (value: any): value is Date => {
	let result: boolean
	if (typeof value == "string") {
		const followPattern = /^\d{4}-\d{2}-\d{2}$/.test(value)
		const month = getMonth(value)
		const validMonth = month >= 1 && month <= 12
		const maxDays = daysInMonth(value)
		const day = getDay(value)
		const validDay = day >= 1 && day <= maxDays
		result = followPattern && validMonth && validDay
	} else {
		result = false
	}
	return result
})

	
	
export function daysInMonth(date: Date): number {
	let result: number
	const year = getYear(date)
	const month = getMonth(date)
	if (month == 2) {
		const isLeapYear = year % 4 === 0 && (year % 100 !== 0 || year % 400 === 0)
		result = isLeapYear ? 29 : 28
	} else {
		result = { 1: 31, 3: 31, 4: 30, 5: 31, 6: 30, 7: 31, 8: 31, 9: 30, 10: 31, 11: 30, 12: 31 }[month] as number
	}
	return result
}
```